src/main.c

```diff
@@ -49,6 +59,56 @@ static void softReset_button_callback(struct device *port, struct gpio_callback
 	soft_reboot();
 }
 
+void cdc_acm_descriptor_callback (cdc_acm_cfg_t *cfg)
+{
+	customer_data_t *otp_data = (customer_data_t *)CUSTOMER_DATA_ADDRESS;
+
+	memset(cfg->vendor_string, '\0', 32);
+	memset(cfg->product_string, '\0', 32);
+	memset(cfg->serial_string, '\0', 32);
+
+	//read the USB descriptors from OTP and use some default values if it has not been properly populated
+	if((otp_data->vendor_name_len>0) && (otp_data->vendor_name_len<=32))
```

Comment:
I see three duplicate blocks of code here. How about you write a function, and in there you can also take care of the null-termination?

for example

    void copy_string_descriptor (char *dest, uint8_t *source, uint32_t len, uint32_t max)
    {
        if (len >= max) {
            len = max - 1;
        }

        if (len > 0) {
            memcpy(dest, src, len);
            dest[len] = '\0';
        }
    }

Then, you can remove those memset calls, and you can replace those three blocks with

    copy_string_descriptor(cfg->vendor_string, otp_data->vendor_name, otp_data->vendor_name_len,
                                         sizeof(cfg->vendor_string));

    copy_string_descriptor(cfg->product_string, otp_data->board_name, otp_data->board_name_len,
                                          sizeof(cfg->product_string));

    copy_string_descriptor(cfg->serial_string, otp_data->product_sn, otp_data->product_sn_len,
                                          sizeof(cfg->serial_string));